Unity-Technologies/com.unity.formats.alembic
# [ABC-312] Fix Face scope constant attributes

## Purpose of this PR

**Ticket/Jira #:[ABC-312](https://jira.unity3d.com/browse/ABC-312)**


When exporting the alembic with face scope, the custom attributes aren't processed as intended. This PR addresses this issue by introducing a method that uses the face count to produce an index array for the attributes.

## Testing
**Functional Testing status:**
- Manual testing : 

  - Open a Unity project with the Alembic package installed 
  - Import [attributeScopeCubes.unitypackage](https://jira.unity3d.com/secure/attachment/477157/477157_attributeScopeCubes.unitypackage) package into the project
  - Instantiate the prefab `attributeScopeCubes` into a scene

Notice how the "cube_face" is now colored, each face with a color (this is the same result obtained when importing the cubes in Houdini). 

- Tests added : 

new `MeshSchema `test class added : 
    - `VertexRgb_AreProcessedCorrectlyForScope` 
    - `VertexRgba_AreProcessedCorrectlyForScope`
    - `VertexUV_AreProcessedCorrectlyForScope` 
    - `VertexNormals_AreProcessedCorrectlyForScope` 

along with the missing data to test rgba, uv and normals. 

| rgb | rgba | uv |
| :--- | :--- | :--- |
| <img src="https://github.com/Unity-Technologies/com.unity.formats.alembic/assets/104858438/d0e17a80-a559-40b5-af45-c2a4fe89e6a2" width="125" height="200"> |<img src="https://github.com/Unity-Technologies/com.unity.formats.alembic/assets/104858438/783d43cc-af77-41ae-9f8c-7b9285f466e5" width="125" height="200">  | <img src="https://github.com/Unity-Technologies/com.unity.formats.alembic/assets/104858438/c3b67e78-6b46-4f9b-81f8-520420c13fb8" width="125" height="200">  |

| Normals |  
| :---  |
|<img src="https://github.com/Unity-Technologies/com.unity.formats.alembic/assets/104858438/75d31ec3-a344-4e15-ac7d-a4828e4ebc50" width="300" height="100"><img src="https://github.com/Unity-Technologies/com.unity.formats.alembic/assets/104858438/f4cecd22-427e-4433-b194-511632b64827" width="300" height="100"> |  



**Performance Testing status:** N/A



## Overall Product Risks


**Complexity:**
Low 

**Halo Effect:**
Minimal 

## Documentation & UX Writing



| User facing text to review |  | Details |
| :--- | :--- | :--- |
| User interface |N  |  |
| Public API docs | N |  |
| Changelog | Y |  |



| Documentation halo effect |  | Jira link |
| :--- | :--- | :--- |
| User manual |  |  |
| Other docs |  |  |

## Additional information

**Note to reviewers:**



**Reminder:**

- [ ] Add entry in CHANGELOG.md
